backend/cmm_helpers.ml

```diff
@@ -3498,8 +3419,7 @@ let arrayset_safe skind arg1 arg2 arg3 dbg =
       bind "newval" arg3 (fun newval ->
           bind "index" arg2 (fun idx ->
               bind "arr" arg1 (fun arr ->
-                  bind "header" (get_header_without_profinfo arr dbg)
-                    (fun hdr ->
+                  bind "header" (get_header_masked arr dbg) (fun hdr ->
```

Comment:
Actually it seems like the existing flambda-backend/main branch differs from tip-jst in ocaml-jst in this regard.  In the latter the ignore_high_bit_int call is in the first argument to "bind".  I've changed it here so that is the case.  This also seems to match 5.x upstream (tip-5 on ocaml-jst).